uprotocol_vsomeip/vsomeip_utransport.py

```diff
@@ -0,0 +1,439 @@
+        self._multicast = multicast
+
+        self._lock = threading.Lock()
+
+        # Get structure and details from template to create configuration
+        if not self._configuration:
+            with open(
+                    os.path.join(os.path.realpath(os.path.dirname(__file__)), 'templates', 'vsomeip_template.json'),
+                    "r",
+                    encoding='utf-8'
+            ) as handle:
+                self._configuration = json.load(handle)
+
+        self._start_services()
+
+    def _start_services(self):
+        """
+        Instantiate all COVESA Services
+        """
+        services = VsomeipHelper.get_services_info()
+        if not services:
+            raise ValueError("Set Services info using VsomeipHelper Class")
+
+        service_instances = []
+        instance_id = 0x1111  # todo: uri.resource.id? what should be the instance id?
+        # version = (entity.version_major if entity.version_major else 0x00,
+        #            entity.version_minor if entity.version_minor else 0x00)
+        version = (0x00, 0x00)  # until changed in sdv autoloader to consider service major minor versions
+        ip_addr = "127.0.0.1"  # todo?: ipaddress.IPv4Address(uri.authority.ip)
+        if is_windows:  # note: vsomeip needs actual address not localhost
+            ip_addr = str(socket.gethostbyname(socket.gethostname()))
+        with self._lock:
+            self._configuration["unicast"] = str(ip_addr)
+            self._configuration["service-discovery"]["multicast"] = str(
+                self._multicast[0])
+            self._configuration["service-discovery"]["port"] = str(
+                self._multicast[1])
+
+            for service_name, service_id in services.items():
+                service_name = VsomeipHelper.replace_special_chars(
+                    service_name
+                ) + '_' + VsomeipTransport.VSOMEIPType.SERVICE.value
+                if service_name not in self._instances:
+                    self._configuration["applications"].append({
+                        'id': str(len(self._instances)),
+                        'name': service_name
+                    })
+
+                    self._configuration["services"].append({
+                        'instance': str(instance_id),
+                        'service': str(service_id),
+                    })
+
+                    instance = vsomeip.SOMEIP(
+                        name=service_name,
+                        id=service_id,
+                        instance=instance_id,
+                        configuration=self._configuration,
+                        version=version)
+                    service_instances.append(instance)
+                    self._instances[service_name] = instance
+
+            for service in service_instances:
+                service.create()
+                service.offer()
+                service.start()
+
+    def _get_instance(self, entity: UEntity,
```

Comment:
_start_services() is for someone who wants vsomeip to host up a service to connect to (in case of uSimulator will start Mock Services).  _get_instance is to create a client instance on the fly when want to connect to any service. 
We can discuss more in the time I set up.